src/whylogs/core/metrics/confusion_matrix.py

```diff
@@ -0,0 +1,148 @@
+from typing import List, Union
+
+import numpy as np
+from sklearn.utils.multiclass import type_of_target
+
+from whylogs.proto import ScoreMatrixMessage
+from whylogs.core.statistics import NumberTracker
+
+
+SUPPORTED_TYPES = ("binary", "multiclass")
+
+
+class ConfusionMatrix:
+
+    """
+    Confusion Matrix Class to hold labels and matrix data
+
+    Attributes:
+        confusion_matrix (nd.array): Description
+        labels (List[str]): Description
+    """
+
+    def __init__(self, labels: List[str] = None,
+                 prediction_field=None,
+                 target_field=None,
+                 score_field=None):
+        self.prediction_field = prediction_field
+        self.target_field = target_field
+        self.score_field = score_field
+        if labels:
+            self.labels = sorted(labels)
+            num_labels = len(self.labels)
+            self.confusion_matrix = np.empty(
+                [num_labels, num_labels], dtype=object)
+            for each_ind_i in range(num_labels):
+                for each_ind_j in range(num_labels):
+                    self.confusion_matrix[each_ind_i,
+                                          each_ind_j] = NumberTracker()
+        else:
+            self.labels = None
+            self.confusion_matrix = None
+
+    def add(self, predictions: List[Union[str, int, bool]],
```

Comment:
The Java API accepts pretty much everything, including `floats` as long as the comparison matches FY (Java doesn't have union type 😅)